src/Lecture4.hs

```diff
@@ -135,8 +144,28 @@ errors. We will simply return an optional result here.
 -}
 
 parseRow :: String -> Maybe Row
-parseRow = error "TODO"
-
+parseRow s =
+  let
+    v = splitAll ',' s
+  in case v of
+    pS : tS : cS : [] -> do
+      tT <- readMaybe (filter (not . isSpace) tS) :: Maybe TradeType
+      cT <- readMaybe (filter (not . isSpace) cS) :: Maybe Int
+      guard $ cT >= 0
+      Just $ Row pS tT cT
+    _ -> Nothing
```

Comment:
You have an excellent monadic implementation 🔝 

I haven't explained the following during lectures but there's a nice way to write this parsing syntax using `do`-notation, `Monad` instance for `Maybe` and the `MonadFail` desugaring trick 🧙🏻 

You can read more about this trick here:

* https://kowainik.github.io/posts/haskell-mini-patterns#monadfail-sugar

Applying this programming pattern and a few helper functions, the `parseRow` function will look like this:

```suggestion
parseRow s = do
    [pS, tS, cS] <- Just $ splitAll ',' s
    guard $ pS /= ""
    tT <- readMaybe tS
    cT <- readMaybe cS
    guard $ cT >= 0
    Just $ Row pS tT cT
```